helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl

```diff
@@ -75,6 +75,35 @@ receivers:
           endpoint: '`endpoint`:`"prometheus.io/port" in annotations ? annotations["prometheus.io/port"] : 9090`'
       {{- end }}
 
+      # Receivers for collecting k8s control plane metrics.
+      {{- if .Values.autodetect.k8sControlPlane }}
+      smartagent/kubernetes-apiserver:
+        # Except for openshift, Kubernetes distributions typically use the same apiserver pod labels.
+        {{- if eq (include "splunk-otel-collector.distribution" .) "openshift" }}
+        rule: type == "pod" && labels["app"] == "openshift-kube-apiserver" && labels["apiserver"] == "true"
+        {{- else }}
+        rule: type == "pod" && labels["k8s-app"] == "kube-apiserver"
+        {{- end }}
+        config:
+          # If the apiserver config is defined by the user, then use that config here.
+          {{- if .Values.k8sControlPlane.apiserver.config -}}
+          - {{- toYaml $.Values.k8sControlPlane.apiserver.config | nindent 10 }}
+          {{- else }}
+          # If the apiserver config is not defined, then we use this default config.
+          - host: localhost
```

Comment:
`localhost` works because both apiserver and splunk-otel-collector pods run in host network, but we shouldn't rely on this. We should use [pod dns](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#a-aaaa-records-1) or pod ip, I don't know if it's possible to do that with `smartagent/kubernetes-apiserver`. Probably can keep it as is and resolve it in the next iteration using receiver_creator/prometheus.

Reply:
Oh it's already under receiver_creator, so it should be possible with some ``magic`` I believe

Reply:
Yeah receiver creator in conjunction with [smartagent receiver will attempt](https://github.com/signalfx/splunk-otel-collector/blob/13bddf49a5f61b4a9abdc1bdabf23ffca7ec5dd2/internal/receiver/smartagentreceiver/config.go#L142-L180) to set the `host` and  `port` automatically from the passed endpoint, and the `host` will be [the pod ip by default](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/772f39c9815f1df5e7943750cd42e7a95994bc29/extension/observer/k8sobserver/pod_endpoint.go#L47). Looks like port would probably not get set, so keeping that may be needed.
Does API server run in a pod or directly on the host? If it's a container running in a pod, maybe you could change the rule to use `type == "port"` since each pod will [create this endpoint type](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/772f39c9815f1df5e7943750cd42e7a95994bc29/extension/observer/k8sobserver/pod_endpoint.go#L60) for each of it's running containers.

Reply:
I was able to use type == "port" successfully, so I dropped the extra host and port configs.